RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift

```diff
@@ -414,8 +475,34 @@ public class RileyLinkDeviceTableViewController: UITableViewController {
             }
 
             showViewController(vc, sender: indexPath)
-        case .Device, .Pump:
+        case .Pump:
             break
         }
     }
+
+    // MARK: - TextFieldTableViewControllerDelegate
+
+    func textFieldTableViewControllerDidReturn(controller: TextFieldTableViewController) {
+        navigationController?.popViewControllerAnimated(true)
+    }
+
+    func textFieldTableViewControllerDidEndEditing(controller: TextFieldTableViewController) {
+        if let indexPath = tableView.indexPathForSelectedRow {
+            switch Section(rawValue: indexPath.section)! {
+            case .Device:
+                switch DeviceRow(rawValue: indexPath.row)! {
+                case .CustomName:
+                    device.setCustomName(controller.value!)
+                default:
+                    break
+                }
+            default:
+                break
+
+            }
+
+            tableView.reloadData()
```

Comment:
No need to broadly reload here, since you're already updating the display after the notification comes in.